### ISSUE TITLE
Grappler fails for TF reference variables

### ISSUE THREAD

**Issue by dsmilkov** [tensorflow/tfjs#1569]
This issues tracks our dependency on TF nightly and should be closed when tf nightly fixes reference variables and we update our python requirements to that nightly version.

Context:
Grappler (used by the tfjs-converter) fails when there are TF v1 reference variables, but there is a temporary work around: Add following line to your python script:
```py
tf.enable_resource_variables()
```

This would use resources variable instead of reference variable.

**Comment by pyu10055** [comment]
@dsmilkov FYI there is an internal TF bug tracking this.

**Comment by dsmilkov** [comment]
Sounds good. This issue tracks that we update our python dependency nightly version when they fix it.

**Comment by csaroff** [comment]
@dsmilkov I ran into this issue when trying to use tensorflowjs_converter to convert a model that I retrained using the tensorflow object detection api.

```bash
tensorflowjs_converter --input_format=tf_saved_model --output_format=tfjs_graph_model models/model/export/Servo/1558145402/ tfjs_model
```

```
2019-05-18 11:52:06.858446: E tensorflow/core/grappler/grappler_item_builder.cc:637] Init node global_step/Assign doesn't exist in graph
Traceback (most recent call last):
  File "/home/csaroff/anaconda3/envs/railroad-vott-tf/bin/tensorflowjs_converter", line 10, in <module>
# Copyright 2017 The TensorFlow Authors. All Rights Reserved.
    sys.exit(main())
  File "/home/csaroff/anaconda3/envs/railroad-vott-tf/lib/python3.7/site-packages/tensorflowjs/converters/converter.py", line 448, in main
    strip_debug_ops=FLAGS.strip_debug_ops)
  File "/home/csaroff/anaconda3/envs/railroad-vott-tf/lib/python3.7/site-packages/tensorflowjs/converters/tf_saved_model_conversion_v2.py", line 276, in convert_tf_saved_model
    concrete_func)
  File "/home/csaroff/anaconda3/envs/railroad-vott-tf/lib/python3.7/site-packages/tensorflow/python/framework/convert_to_constants.py", line 99, in convert_variables_to_constants_v2
    graph_def = _run_inline_graph_optimization(func)
  File "/home/csaroff/anaconda3/envs/railroad-vott-tf/lib/python3.7/site-packages/tensorflow/python/framework/convert_to_constants.py", line 57, in _run_inline_graph_optimization
    return tf_optimizer.OptimizeGraph(config, meta_graph)
  File "/home/csaroff/anaconda3/envs/railroad-vott-tf/lib/python3.7/site-packages/tensorflow/python/grappler/tf_optimizer.py", line 41, in OptimizeGraph
    verbose, graph_id)
tensorflow.python.framework.errors_impl.InvalidArgumentError: Failed to import metagraph, check error log for more info.
```


I tried adding your workaround code(`tf.enable_resource_variables()`) directly to `packages/tensorflow/python/grappler/tf_optimizer.py`

I'm seeing `AttributeError: module 'tensorflow' has no attribute 'enable_resource_variables'`.

I'm kind of a novice at python package management.  `pip freeze` shows 

```
tensorflow==1.13.1
tf-nightly-2.0-preview==2.0.0.dev20190518
```

It looks like the 2.x nightly preview is loading in though.

```bash
python -c "import tensorflow as tf; print(tf.__version__)"
```
Gives
```bash
2.0.0-dev20190518
```

Any insight here would be super helpful if you get a chance!

**Comment by pyu10055** [comment]
this is fixed upstream in the latest TF 1.15 and 2.0
